### PR TITLE
#632 Support state dependent parameters

### PR DESCRIPTION
**Pull request by cozminvelciu**
Add Crossview support for state dependent parameters.

### CODE REVIEW

**Comment by cozminvelciu** [comment]
Not particularly pleased of the `PrettifyBodyHeader` method, but it works. ¯\\_(ツ)_/¯

File: CDP4CrossViewEditor/Generator/CrossviewSheetConstants.cs [hunk 1]
```diff
@@ -38,10 +38,10 @@ internal static class CrossviewSheetConstants
         internal const int FixedColumns = 5;
 
         /// <summary>
-        /// The number of header nested "layers": ParamterType, MeasurementUnit.
-        /// NOTE: There might be fewer actual header rows if all values on a layer are missing.
+        /// The number of header nested "layers":
+        /// ParamterType -> ActualFiniteStateList -> ActualFiniteState -> MeasurementUnit.
```
Comment (lxatrhea) [inline CDP4CrossViewEditor/Generator/CrossviewSheetConstants.cs:42]: small typo paramterType

**Review by lxatrhea — APPROVED** [review]

File: CDP4CrossViewEditor/Generator/CrossviewSheetConstants.cs (hunk 1 above)
Comment (cozminvelciu) [reply CDP4CrossViewEditor/Generator/CrossviewSheetConstants.cs]: Fixed

**Review by ahmedatrhea — APPROVED** [review]

File: CDP4CrossViewEditor/Assemblers/CrossviewArrayAssembler.cs [hunk 2]
```diff
@@ -64,10 +60,11 @@ public sealed class CrossviewArrayAssembler
         private IEnumerable<ParameterType> parameterTypes;
 
         /// <summary>
-        /// The header structure mapping parameters to indices
+        /// The header structure mapping parameter layer short names to indices:
+        /// ParamterType -> ActualFiniteStateList -> ActualFiniteState -> MeasurementUnit -> index
```
Comment (adrianchivu) [inline CDP4CrossViewEditor/Assemblers/CrossviewArrayAssembler.cs:64]: Small typo.

**Review by adrianchivu — APPROVED** [review]
Looks fine, didn't have the chance to see it in action locally.

**Pull request merged**